### ISSUE TITLE
[Docs] `torch.nn.Dropout2d` accepts 2D inputs, but it's not listed

### ISSUE THREAD

**Issue by krshrimali** [pytorch/pytorch#67892]
## 📚 Documentation

[`torch.nn.Dropout2d`](https://pytorch.org/docs/master/generated/torch.nn.Dropout2d.html?highlight=dropout2d#torch.nn.Dropout2d) documentation mentions supported shapes are `(N, C, H, W) or (C, H, W)` for inputs, but input shape of `(H, W)` is also supported.

IMO, the documentation should be revised to mention the input and output shapes for `(H, W)`.

```python
# A 2D tensor is a valid input for Dropout2d
In [11]: tensor = torch.randn((3, 4), device='cpu', dtype=torch.float32)
In [12]: dropout2d = torch.nn.Dropout2d(p=0.5)

In [13]: dropout2d(tensor)
Out[13]:
tensor([[-0.1026, -0.0000, -0.0000, -0.0000],
        [-1.5647,  0.0000, -0.0000, -0.5820],
        [-0.0000, -3.2080,  0.1164, -3.6780]])
```

Note: If we decide it's worth fixing, I can send this fix along with the OpInfo here: https://github.com/pytorch/pytorch/pull/67891.

cc: @jbschlosser @zou3519

cc @brianjo @mruberry @albanD @jbschlosser @walterddr

**Comment by jbschlosser** [comment]
Hey @krshrimali, thanks for the report! Sure, we'd accept a PR updating the docs to indicate shape `(H, W)` is supported.

**Comment by jbschlosser** [comment]
Update: with the addition of no-batch-dim support for dropout2d, we've now deprecated passing 2D inputs to it. As such, we don't want to document this sort of usage.
